File: sdk/spring/src/main/java/io/swagger/model/ApprovalRule.java
package io.swagger.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import org.joda.time.LocalDate;





@javax.annotation.Generated(value = "class io.swagger.codegen.languages.SpringCodegen", date = "2016-07-26T21:59:26.456Z")

public class ApprovalRule   {
  
  private String ID = null;
  private String name = null;
  private String description = null;
  private String approvingGroupID = null;
  private String ruleExpression = null;
  private String scope = null;
  private String scopeTimeUnit = null;
  private Integer scopeTimeNumber = null;
  private LocalDate scopeStartDate = null;
  private String expireAfterTimeUnit = null;
  private Integer expireAfterNumber = null;
  private Boolean approveOnExpire = null;
  private Object xp = null;

  /**
   **/
  public ApprovalRule ID(String ID) {
    this.ID = ID;
    return this;
  }

  
  @ApiModelProperty(value = "")
  @JsonProperty("ID")
  public String getID() {
    return ID;
  }
  public void setID(String ID) {
    this.ID = ID;
  }

  /**
   **/
  public ApprovalRule name(String name) {
    this.name = name;
    return this;
  }

  
  @ApiModelProperty(value = "")
  @JsonProperty("Name")
  public String getName() {
    return name;
  }
  public void setName(String name) {
    this.name = name;
  }

  /**
   **/
  public ApprovalRule description(String description) {
    this.description = description;
    return this;
  }

  
  @ApiModelProperty(value = "")
  @JsonProperty("Description")
  public String getDescription() {
    return description;
  }
  public void setDescription(String description) {
    this.description = description;
  }

  /**
   **/
  public ApprovalRule approvingGroupID(String approvingGroupID) {
    this.approvingGroupID = approvingGroupID;
    return this;
  }

  
  @ApiModelProperty(value = "")
  @JsonProperty("ApprovingGroupID")
  public String getApprovingGroupID() {
    return approvingGroupID;
  }
  public void setApprovingGroupID(String approvingGroupID) {
    this.approvingGroupID = approvingGroupID;
  }

  /**
   **/
  public ApprovalRule ruleExpression(String ruleExpression) {
    this.ruleExpression = ruleExpression;
    return this;
  }

  
  @ApiModelProperty(value = "")
  @JsonProperty("RuleExpression")
  public String getRuleExpression() {
    return ruleExpression;
  }
  public void setRuleExpression(String ruleExpression) {
    this.ruleExpression = ruleExpression;
  }

  /**
   **/
  public ApprovalRule scope(String scope) {
    this.scope = scope;
    return this;
  }

  
  @ApiModelProperty(value = "")
  @JsonProperty("Scope")
  public String getScope() {
    return scope;
  }
  public void setScope(String scope) {
    this.scope = scope;
  }

  /**
   **/
  public ApprovalRule scopeTimeUnit(String scopeTimeUnit) {
    this.scopeTimeUnit = scopeTimeUnit;
    return this;
  }

  
  @ApiModelProperty(value = "")
  @JsonProperty("ScopeTimeUnit")
  public String getScopeTimeUnit() {
    return scopeTimeUnit;
  }
  public void setScopeTimeUnit(String scopeTimeUnit) {
    this.scopeTimeUnit = scopeTimeUnit;
  }

  /**
   **/
  public ApprovalRule scopeTimeNumber(Integer scopeTimeNumber) {
    this.scopeTimeNumber = scopeTimeNumber;
    return this;
  }

  
  @ApiModelProperty(value = "")
  @JsonProperty("ScopeTimeNumber")
  public Integer getScopeTimeNumber() {
    return scopeTimeNumber;
  }
  public void setScopeTimeNumber(Integer scopeTimeNumber) {
    this.scopeTimeNumber = scopeTimeNumber;
  }

  /**
   **/
  public ApprovalRule scopeStartDate(LocalDate scopeStartDate) {
    this.scopeStartDate = scopeStartDate;
    return this;
  }

  
  @ApiModelProperty(value = "")
  @JsonProperty("ScopeStartDate")
  public LocalDate getScopeStartDate() {
    return scopeStartDate;
  }
  public void setScopeStartDate(LocalDate scopeStartDate) {
    this.scopeStartDate = scopeStartDate;
  }

  /**
   **/
  public ApprovalRule expireAfterTimeUnit(String expireAfterTimeUnit) {
    this.expireAfterTimeUnit = expireAfterTimeUnit;
    return this;
  }

  
  @ApiModelProperty(value = "")
  @JsonProperty("ExpireAfterTimeUnit")
  public String getExpireAfterTimeUnit() {
    return expireAfterTimeUnit;
  }
  public void setExpireAfterTimeUnit(String expireAfterTimeUnit) {
    this.expireAfterTimeUnit = expireAfterTimeUnit;
  }

  /**
   **/
  public ApprovalRule expireAfterNumber(Integer expireAfterNumber) {
    this.expireAfterNumber = expireAfterNumber;
    return this;
  }

  
  @ApiModelProperty(value = "")
  @JsonProperty("ExpireAfterNumber")
  public Integer getExpireAfterNumber() {
    return expireAfterNumber;
  }
  public void setExpireAfterNumber(Integer expireAfterNumber) {
    this.expireAfterNumber = expireAfterNumber;
  }

  /**
   **/
  public ApprovalRule approveOnExpire(Boolean approveOnExpire) {
    this.approveOnExpire = approveOnExpire;
    return this;
  }

  
  @ApiModelProperty(value = "")
  @JsonProperty("ApproveOnExpire")
  public Boolean getApproveOnExpire() {
    return approveOnExpire;
  }
  public void setApproveOnExpire(Boolean approveOnExpire) {
    this.approveOnExpire = approveOnExpire;
  }

  /**
   **/
  public ApprovalRule xp(Object xp) {
    this.xp = xp;
    return this;
  }

  
  @ApiModelProperty(value = "")
  @JsonProperty("xp")
  public Object getXp() {
    return xp;
  }
  public void setXp(Object xp) {
    this.xp = xp;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    ApprovalRule approvalRule = (ApprovalRule) o;
    return Objects.equals(ID, approvalRule.ID) &&
        Objects.equals(name, approvalRule.name) &&
        Objects.equals(description, approvalRule.description) &&
        Objects.equals(approvingGroupID, approvalRule.approvingGroupID) &&
        Objects.equals(ruleExpression, approvalRule.ruleExpression) &&
        Objects.equals(scope, approvalRule.scope) &&
        Objects.equals(scopeTimeUnit, approvalRule.scopeTimeUnit) &&
        Objects.equals(scopeTimeNumber, approvalRule.scopeTimeNumber) &&
        Objects.equals(scopeStartDate, approvalRule.scopeStartDate) &&
        Objects.equals(expireAfterTimeUnit, approvalRule.expireAfterTimeUnit) &&
        Objects.equals(expireAfterNumber, approvalRule.expireAfterNumber) &&
        Objects.equals(approveOnExpire, approvalRule.approveOnExpire) &&
        Objects.equals(xp, approvalRule.xp);
  }

  @Override
  public int hashCode() {
    return Objects.hash(ID, name, description, approvingGroupID, ruleExpression, scope, scopeTimeUnit, scopeTimeNumber, scopeStartDate, expireAfterTimeUnit, expireAfterNumber, approveOnExpire, xp);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class ApprovalRule {\n");
    
    sb.append("    ID: ").append(toIndentedString(ID)).append("\n");
    sb.append("    name: ").append(toIndentedString(name)).append("\n");
    sb.append("    description: ").append(toIndentedString(description)).append("\n");
    sb.append("    approvingGroupID: ").append(toIndentedString(approvingGroupID)).append("\n");
    sb.append("    ruleExpression: ").append(toIndentedString(ruleExpression)).append("\n");
    sb.append("    scope: ").append(toIndentedString(scope)).append("\n");
    sb.append("    scopeTimeUnit: ").append(toIndentedString(scopeTimeUnit)).append("\n");
    sb.append("    scopeTimeNumber: ").append(toIndentedString(scopeTimeNumber)).append("\n");
    sb.append("    scopeStartDate: ").append(toIndentedString(scopeStartDate)).append("\n");
    sb.append("    expireAfterTimeUnit: ").append(toIndentedString(expireAfterTimeUnit)).append("\n");
    sb.append("    expireAfterNumber: ").append(toIndentedString(expireAfterNumber)).append("\n");
    sb.append("    approveOnExpire: ").append(toIndentedString(approveOnExpire)).append("\n");
    sb.append("    xp: ").append(toIndentedString(xp)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: sdk/CsharpDotNet2/src/main/CsharpDotNet2/OrderCloud/Model/ApprovalRule.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace OrderCloud.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class ApprovalRule {
    /// <summary>
    /// Gets or Sets ID
    /// </summary>
    [DataMember(Name="ID", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "ID")]
    public string ID { get; set; }

    /// <summary>
    /// Gets or Sets Name
    /// </summary>
    [DataMember(Name="Name", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "Name")]
    public string Name { get; set; }

    /// <summary>
    /// Gets or Sets Description
    /// </summary>
    [DataMember(Name="Description", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "Description")]
    public string Description { get; set; }

    /// <summary>
    /// Gets or Sets ApprovingGroupID
    /// </summary>
    [DataMember(Name="ApprovingGroupID", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "ApprovingGroupID")]
    public string ApprovingGroupID { get; set; }

    /// <summary>
    /// Gets or Sets RuleExpression
    /// </summary>
    [DataMember(Name="RuleExpression", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "RuleExpression")]
    public string RuleExpression { get; set; }

    /// <summary>
    /// Gets or Sets Scope
    /// </summary>
    [DataMember(Name="Scope", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "Scope")]
    public string Scope { get; set; }

    /// <summary>
    /// Gets or Sets ScopeTimeUnit
    /// </summary>
    [DataMember(Name="ScopeTimeUnit", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "ScopeTimeUnit")]
    public string ScopeTimeUnit { get; set; }

    /// <summary>
    /// Gets or Sets ScopeTimeNumber
    /// </summary>
    [DataMember(Name="ScopeTimeNumber", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "ScopeTimeNumber")]
    public int? ScopeTimeNumber { get; set; }

    /// <summary>
    /// Gets or Sets ScopeStartDate
    /// </summary>
    [DataMember(Name="ScopeStartDate", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "ScopeStartDate")]
    public DateTime? ScopeStartDate { get; set; }

    /// <summary>
    /// Gets or Sets ExpireAfterTimeUnit
    /// </summary>
    [DataMember(Name="ExpireAfterTimeUnit", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "ExpireAfterTimeUnit")]
    public string ExpireAfterTimeUnit { get; set; }

    /// <summary>
    /// Gets or Sets ExpireAfterNumber
    /// </summary>
    [DataMember(Name="ExpireAfterNumber", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "ExpireAfterNumber")]
    public int? ExpireAfterNumber { get; set; }

    /// <summary>
    /// Gets or Sets ApproveOnExpire
    /// </summary>
    [DataMember(Name="ApproveOnExpire", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "ApproveOnExpire")]
    public bool? ApproveOnExpire { get; set; }

    /// <summary>
    /// Gets or Sets Xp
    /// </summary>
    [DataMember(Name="xp", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "xp")]
    public Object Xp { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class ApprovalRule {\n");
      sb.Append("  ID: ").Append(ID).Append("\n");
      sb.Append("  Name: ").Append(Name).Append("\n");
      sb.Append("  Description: ").Append(Description).Append("\n");
      sb.Append("  ApprovingGroupID: ").Append(ApprovingGroupID).Append("\n");
      sb.Append("  RuleExpression: ").Append(RuleExpression).Append("\n");
      sb.Append("  Scope: ").Append(Scope).Append("\n");
      sb.Append("  ScopeTimeUnit: ").Append(ScopeTimeUnit).Append("\n");
      sb.Append("  ScopeTimeNumber: ").Append(ScopeTimeNumber).Append("\n");
      sb.Append("  ScopeStartDate: ").Append(ScopeStartDate).Append("\n");
      sb.Append("  ExpireAfterTimeUnit: ").Append(ExpireAfterTimeUnit).Append("\n");
      sb.Append("  ExpireAfterNumber: ").Append(ExpireAfterNumber).Append("\n");
      sb.Append("  ApproveOnExpire: ").Append(ApproveOnExpire).Append("\n");
      sb.Append("  Xp: ").Append(Xp).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}
